internal/pkg/otel/run.go

```diff
@@ -33,6 +35,10 @@ func Run(ctx context.Context, stop chan bool, configFiles []string) error {
 		return err
 	}
 
+	if err := ensureRegistryExists(); err != nil {
+		return fmt.Errorf("error while creating registry: %w", err)
+	}
+
```

Comment:
Shouldn't we check and inject the `storage` extension settings here? It may be conditional to a cli flag or some env variable or something else but I doubt very man people are gonna run the elastic otel collector with our sample config file and without that this change ensures only that a given dir exists/is created...